oauthproxy.go

```diff
@@ -371,8 +374,12 @@ func buildSessionChain(opts *options.Options, provider providers.Provider, sessi
 	chain := alice.New()
 
 	if opts.SkipJwtBearerTokens {
-		sessionLoaders := []middlewareapi.TokenToSessionFunc{
-			provider.CreateSessionFromToken,
+
+		sessionLoaders := []middlewareapi.TokenToSessionFunc{}
+		if opts.IntrospectToken {
+			sessionLoaders = append(sessionLoaders, provider.CreateSessionFromIntrospectedToken)
+		} else {
+			sessionLoaders = append(sessionLoaders, provider.CreateSessionFromToken)
```

Comment:
We really shouldn't force token introspection for `id_tokens` as well!? It might be a valid use case to have the possibility to still use the `CreateSessionFromToken` logic for `id_tokens` and the new `CreateSessionFromIntrospectedToken` for `access_tokens`

Maybe something like this:
```suggestion
		sessionLoaders = []middlewareapi.TokenToSessionFunc{}
		
		sessionLoaders = append(sessionLoaders, provider.CreateSessionFromToken)
		
		if opts.IntrospectToken {
			sessionLoaders = append(sessionLoaders, provider.CreateSessionFromIntrospectedToken)
```

Reply:
@JoelSpeed @NickMeves what is your take on this? Should we keep both and introduce a new flag called `skip-bearer-tokens` and deprecate `skip-jwt-bearer-tokens` as mentioned below?